### DISCUSSION TITLE
low "Number of groups accepted" ratio in UMI correction

### DISCUSSION THREAD

**Discussion by bigcat1001** [milaboratory/mixcr#1008] [Q&A] [answered]
Hi,
     When I used UMI to correct clones, I found that very few clones were finally obtained. After reading the 'Rep-Seq libraries' section, I think it may be that there are few clones in the sample.But in the process, I found a problem that was not described in the troubleshooting:Number of groups accepted is only 32.74% . But in the report, I can't see why so many UMIs are filtered.
Is this also caused by the low clone content in the sample? Or there are some parameters that can improve the acceptance rate here？

> Analysis time: 50.93s
Time spent in correction: 11.69s
UMI input diversity: 14473
UMI output diversity: 1564 (10.81%)
UMI input reads: 968826
UMI output count: 966930 (99.8%)
UMI mean reads per tag: 66.94
UMI input core diversity: 3069 (21.21%)
UMI input core reads: 943220 (97.36%)
UMI directly corrected diversity: 11049 (76.34%)
UMI directly corrected reads: 33482 (3.46%)
UMI diversity filtered by tag quality: 1860 (12.85%)
UMI reads filtered by tag quality: 1896 (0.2%)
UMI diversity filtered by whitelist: 0 (0%)
UMI recursively corrected: 2669
Number of output records: 965231 (99.63%)
Filter report:
  Number of groups: 1564
  Number of groups accepted: 512 (32.74%)
  Total records weight: 966930.0
  Records weight accepted: 965231.0 (99.82%)
  Operator #0:
    Effective threshold: 18.0

**Comment by mizraelson** [comment, marked as answer]
Hi,
So the two major numbers here are : 
Number of groups accepted: 512 (32.74%)
Records weight accepted: 965231.0 (99.82%)

This shows that most of the reads passed the filter (99.82 %), the rest of the reads (0.18%) contributed to the artificial diversity and were filtered out.

The filter depends on the preset, but most likely, if you used one of default amplicon presets, the applied filtering used Otsu algorithm witch automatically sets the reads per UMI threshold. 

From the line:

Effective threshold: 18.0

We see that all UMIs covered by less than 18 reads were filtered.

You can also visualize this distribution and the threshold using:

`mixcr exportQc tags sample.refined.vdjca tags.pdf`

Where,
sample.refined.vdjca is the file with alignments after refinement and filtering.

In a common case in tags.pdf you should see two peaks of reads/umi distribution and a red line representing the threshold separating these two peaks.

**Reply by bigcat1001** [reply]
Hi,
   Me again.Can I manually adjust the threshold size? This may increase the number of UMI Groups in the case of over-sequencing.

**Reply by mizraelson** [reply]
Hi, so if you are using analyze command you can add the parameters you see in the example below:

```

mixcr analyze takara-mouse-tcr-cdr3 \
-Massemble.consensusAssemblerParameters.assembler.minRecordsPerConsensus=1 \ 
-MrefineTagsAndSort.parameters.postFilter=null \
input_R1.fastq.gz \
input_R2.fastq.gz \
output
```

Where, 1 is the UMI threshold.

In the future release we will add a more user friendly parameter.

I recommend to check the distribution first to make sure you are not gonna introduce artificial Umi diversity by lowering the threshold.

**Comment by bigcat1001** [comment]
Thanks,now i understand.
